pkgs/by-name/ko/kotatsu-dl/package.nix

```diff
@@ -0,0 +1,41 @@
+{ pkgs }:
+
+pkgs.stdenv.mkDerivation rec {
+  pname = "kotatsu-dl";
+  version = "0.5";
+
+  src = pkgs.fetchurl {
+    url = "https://github.com/KotatsuApp/kotatsu-dl/releases/download/v${version}/kotatsu-dl.jar";
+    sha256 = "sha256-GI/PLt8wbVs2R78PbQg/9/Rea33j6CFQjuqhYIQGimM=";
+  };
+
+  # Skip the unpackPhase since it's just a JAR file, not an archive
+  phases = [ "installPhase" ];
+
+  # Java Runtime
+  buildInputs = [ pkgs.openjdk ];
+  propagatedBuildInputs = [ pkgs.openjdk ];
+
+  installPhase = ''
```

Comment:
Start and end every manually-defined main phase with `runHook pre<phaseName>` and `runHook post<phaseName>`.
So here it would be 
```nix
installPhase = ''
  runHook preInstall
  # Installing...
  runHook postInstall
''
```